src/gui/processing/qgsprocessingoutputdestinationwidget.cpp

```diff
@@ -177,7 +177,15 @@ QVariant QgsProcessingLayerOutputDestinationWidget::value() const
     {
       // output name does not include a folder - use default
       QString defaultFolder = settings.value( QStringLiteral( "/Processing/Configuration/OUTPUTS_FOLDER" ), QStringLiteral( "%1/processing" ).arg( QDir::homePath() ) ).toString();
-      key = QDir( defaultFolder ).filePath( key );
+      QDir destDir( defaultFolder );
+      if ( !destDir.exists() )
+      {
+        if ( !QDir().mkpath( defaultFolder ) )
```

Comment:
Maybe we can join this checks?
```suggestion
      QDir destDir( defaultFolder );
      if ( !destDir.exists() && !QDir().mkpath( defaultFolder ) )
```